### ISSUE TITLE
Regression: groupBySource doesn't work correctly for gtest

### ISSUE THREAD

**Issue by stefanhaller** [matepek/vscode-catch2-test-adapter#310]
1. Use the following test file:
```
#include "gtest/gtest.h"

TEST(T, t1)
{
  EXPECT_TRUE(false);
}

TEST(T, t2)
{
  EXPECT_TRUE(false);
}

int main(int argc, char** argv)
{
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
```

2. Set `"testMate.cpp.test.advancedExecutables"` to
```
    "testMate.cpp.test.advancedExecutables": [
        {
            "pattern": "build/*_test",
            "gtest": {
                "testGrouping": {
                    "groupBySource": {},
                }
            }
        }
    ]
```

3. When initially opening the window (or invoking "C++ TestMate: Reload Tests", the test browser only shows t2, but not t1:

<img width="415" alt="image" src="https://user-images.githubusercontent.com/1225667/144718835-4099feb5-0265-40b6-840c-4c05268d2ac4.png">

4. If you then touch the executable to make TestMate reload it, it shows both tests correctly:

<img width="415" alt="image" src="https://user-images.githubusercontent.com/1225667/144718873-ea1ecc40-811c-4b4e-a0e1-afd8852cb067.png">

- Extension Version: 4.0.9

I suspect that this is a regression caused by 21c98a7bb405; if I comment out [this `if` statement](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/src/AbstractExecutable.ts#L232), it works correctly again (but of course the source file no longer has a URI).

**Comment by matepek** [comment]
might useful:
```
      "gtest": {
        "testGrouping": {
            "groupBySource": {
              "groupByTags": {
                "tagFormat": "${tag}"
              }
            }
        }
      }
```

**Comment by matepek** [comment]
Fixed in v4.0.10
